slackbot_using_azure_openai/service.bal

```diff
@@ -0,0 +1,78 @@
+import ballerina/crypto;
+import ballerina/http;
+import ballerinax/azure.openai.chat;
+
+configurable string openAIToken = ?;
+configurable string serviceUrl = ?;
+configurable string deploymentId = ?;
+configurable string slackSigningSecret = ?;
+
+const API_VERSION = "2023-03-15-preview";
+const MAX_MESSAGES = 25;
+
+final chat:Client azureOpenAI = check new (
+    config = {auth: {apiKey: openAIToken}},
+    serviceUrl = serviceUrl
+);
+
+type ChatMessage record {|
+    string role;
+    string content;
+|};
+
+service /slack on new http:Listener(8080) {
+    map<ChatMessage[]> chatHistory = {};
+
+    resource function post events(http:Request request) returns json|error {
+        if !check verifyRequest(request) {
+            return error("Request verification failed");
+        }
+
+        map<string> requestPayload = check request.getFormParams();
+
+        string? chanelName = requestPayload["channel_name"];
+        string? requestText = requestPayload["text"];
+        if chanelName !is string || requestText !is string {
+            return error("Error in request");
+        }
+
+        ChatMessage[] history = self.chatHistory[chanelName] ?: [];
+        if history.length() == 0 {
+            history = [{role: "system", content: "You are an AI slack bot to assist with user questions."}, {role: "user", content: requestText}];
+        } else {
+            history.push({role: "user", content: requestText});
+        }
+
+        chat:Chat_completions_body chatBody = {messages: history};
+        chat:Inline_response_200 completion = check azureOpenAI->/deployments/[deploymentId]/chat/completions.post(API_VERSION, chatBody);
+
+        chat:Inline_response_200_message? response = completion.choices[0].message;
+        if response !is chat:Inline_response_200_message {
```

Comment:
```suggestion
        if response is () {
```

Reply:
Addressed in [6737695](https://github.com/ballerina-guides/ai-samples/pull/64/commits/6737695630388bbd0350796f53bcd230e7b02993)